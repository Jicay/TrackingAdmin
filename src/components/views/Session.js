import React, {Component} from "react";
import {Link} from "react-router-dom";
import ImageForm from "./ImageForm";
import Home from "./Home";
import Congrats from "./Congrats";
import MouseIndicator from "./MouseIndicator";

export default class Session extends Component {
    constructor(props) {
        super(props);
        const {match} = this.props;

        const sessionId = match.params.sessionId;

        this.eventSource = new EventSource("https://tracking-mouse-back.herokuapp.com/sessions/" + sessionId + "/mouse-positions/stream");

        this.state = {
            page: 'home',
            x: 0,
            y: 0
        };
    }

    componentDidMount() {
        this.eventSource.addEventListener("mouse-position-event", e =>
            this.updateMousePosition(JSON.parse(e.data))
        );
    }

    updateMousePosition(mousePosition) {
        const height = document.getElementById('inner-page').offsetHeight;
        const width = document.getElementById('inner-page').offsetWidth;

        this.setState({page: mousePosition.page, x: mousePosition.x, y: mousePosition.y, height: height, width: width});
    }

    render() {
        const {match} = this.props;
        const {x, y, width, height} = this.state;

        const sessionId = match.params.sessionId;

        return (
            <div className="container">
                <Link to='/'>Retour à la liste des sessions</Link>
                <div>
                    Détail de la session {sessionId}
                </div>
                <div id="remote-page">
                    <MouseIndicator top={y * height} left={x * width}/>
                    <div id="inner-page">
                        {this.renderPage()}
                    </div>
                </div>
            </div>
        );
    }

    renderPage() {
        const {page} = this.state;

        if (page === 'home') {
            return (
                <Home/>
            )
        } else if (page === 'image') {
            return (
                <ImageForm/>
            )
        } else if (page === 'congrats') {
            return (
                <Congrats/>
            )
        } else {
            return null;
        }
    }
}