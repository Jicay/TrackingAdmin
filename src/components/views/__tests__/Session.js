import React from 'react';
import Session from "../Session";
import EventSource from 'eventsourcemock';
import {MemoryRouter} from "react-router";
import * as Enzyme from 'enzyme';
import {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';

Enzyme.configure({adapter: new Adapter()})

describe('Session', () => {
    it('renders correctly when home', () => {
        const match = {
            params: {
                sessionId: 10
            }
        }

        Object.defineProperty(window, 'EventSource', {
            value: EventSource,
        });

        const session = shallow(<Session match={match}/>)

        session.setState({
            page: 'home',
            x: 365,
            y: 128,
            width: 1000,
            height: 1200
        })

        const r = shallow((<MemoryRouter keyLength={0}>{session}</MemoryRouter>))

        const tree = toJSON(r);
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly when image', () => {
        const match = {
            params: {
                sessionId: 10
            }
        }

        Object.defineProperty(window, 'EventSource', {
            value: EventSource,
        });

        const session = shallow(<Session match={match}/>)

        session.setState({
            page: 'image',
            x: 365,
            y: 128,
            width: 1000,
            height: 1200
        })

        const r = shallow((<MemoryRouter keyLength={0}>{session}</MemoryRouter>))

        const tree = toJSON(r);
        expect(tree).toMatchSnapshot();
    });


    it('renders correctly when congrats', () => {
        const match = {
            params: {
                sessionId: 10
            }
        }

        Object.defineProperty(window, 'EventSource', {
            value: EventSource,
        });

        const session = shallow(<Session match={match}/>)

        session.setState({
            page: 'congrats',
            x: 365,
            y: 128,
            width: 1000,
            height: 1200
        })

        const r = shallow((<MemoryRouter keyLength={0}>{session}</MemoryRouter>))

        const tree = toJSON(r);
        expect(tree).toMatchSnapshot();
    });
})