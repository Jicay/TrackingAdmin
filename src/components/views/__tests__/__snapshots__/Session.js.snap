// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Session renders correctly when congrats 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <div
    className="container"
  >
    <Link
      replace={false}
      to="/"
    >
      Retour à la liste des sessions
    </Link>
    <div>
      Détail de la session 
      10
    </div>
    <div
      id="remote-page"
    >
      <MouseIndicator
        left={365000}
        top={153600}
      />
      <div
        id="inner-page"
      >
        <Congrats />
      </div>
    </div>
  </div>
</Router>
`;

exports[`Session renders correctly when home 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <div
    className="container"
  >
    <Link
      replace={false}
      to="/"
    >
      Retour à la liste des sessions
    </Link>
    <div>
      Détail de la session 
      10
    </div>
    <div
      id="remote-page"
    >
      <MouseIndicator
        left={365000}
        top={153600}
      />
      <div
        id="inner-page"
      >
        <Home />
      </div>
    </div>
  </div>
</Router>
`;

exports[`Session renders correctly when image 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <div
    className="container"
  >
    <Link
      replace={false}
      to="/"
    >
      Retour à la liste des sessions
    </Link>
    <div>
      Détail de la session 
      10
    </div>
    <div
      id="remote-page"
    >
      <MouseIndicator
        left={365000}
        top={153600}
      />
      <div
        id="inner-page"
      >
        <ImageForm />
      </div>
    </div>
  </div>
</Router>
`;
